pallets/committee-management/src/lib.rs

```diff
@@ -28,31 +28,57 @@ mod migration;
 mod traits;
 
 use codec::{Decode, Encode};
-use frame_support::traits::StorageVersion;
+use frame_support::{pallet_prelude::Get, traits::StorageVersion};
 pub use manager::SessionAndEraManager;
 pub use migration::PrefixMigration;
 pub use pallet::*;
-use primitives::{BanConfig as BanConfigStruct, BanInfo, SessionValidators};
+use primitives::{BanConfig as BanConfigStruct, BanInfo, SessionValidators, LENIENT_THRESHOLD};
 use scale_info::TypeInfo;
-use sp_std::{collections::btree_map::BTreeMap, default::Default};
+use sp_runtime::Perquintill;
+use sp_std::{collections::btree_map::BTreeMap, default::Default, mem::replace};
 pub use traits::*;
 
 pub type TotalReward = u32;
 #[derive(Decode, Encode, TypeInfo, PartialEq, Eq)]
 pub struct ValidatorTotalRewards<T>(pub BTreeMap<T, TotalReward>);
 
-#[derive(Decode, Encode, TypeInfo)]
-struct CurrentAndNextSessionValidators<T> {
-    pub next: SessionValidators<T>,
-    pub current: SessionValidators<T>,
+#[derive(Decode, Encode, TypeInfo, Default)]
+pub struct ChangingValue<T> {
```

Comment:
I like a lot what you did here, adding the ChangingValue. This is generally a great addition.

However... xD

I'm not so sure for this particular case this is necessary. I mean why don't we hold the lenient_threshold as `Perquintill` and just update it normally? I can see no disadvantage:
- It is simpler to code, and simpler to understand the code
- When using it, we don't need to wait a full session for the new value to kick in. Which in this case might not be a big deal, but still...

Reply:
I thought it might be unfair to others who had different threshold in the era :P

Reply:
Hah, yeah, that's a valid reason :P. I didn't even realize it's changing on era boundaries, not session boundaries. Still, I think it's better to give more control to the "admin" here -- they can choose to do that on era change if they wish :) But normally we will probably want the change with no delay.

Reply:
ok, done